fairlearn/preprocessing/_correlation_remover.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) Vincent D. Warmerdam, Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+
+class CorrelationRemover(BaseEstimator, TransformerMixin):
+    r"""
+     A component that filters out sensitive correlations in a dataset.
+
+     CorrelationRemover applies a linear transformation to the non-sensitive feature columns in order
+     to remove their correlation with the sensitive feature columns while retaining as much information
+     as possible (as measured by the least-squares error).
+
+     Parameters
+     ----------
+     sensitive_feature_ids : list of columns to filter out this can be a sequence of either int
+       ,in the case of numpy, or string, in the case of pandas.
+     alpha : parameter to control how much to filter, for alpha=1.0 we filter out
+       all information while for alpha=0.0 we don't apply any.
+     center : setting to tell if this preprocessing step should center the data for
+       numerical stability
+
+     Notes
+     -----
+
+     This method will change the original dataset by removing all correlation with sensitive values.
+     To describe that mathematically, let's assume in the original dataset :math:`X` we've got a set of
+     sensitive attributes :math:`S` and a set of non-sensitive attributes :math:`Z`. Mathematically this method
+     will be solving the following problem.
+
+     .. math::
+
+        \min _{\mathbf{z}_{1}, \ldots, \mathbf{z}_{n}} \sum_{i=1}^{n}\left\|\mathbf{z}_{i}-\mathbf{x}_{i}\right\|^{2} \\
+        \text{subject to} \\
+        \frac{1}{n} \sum_{i=1}^{n} \mathbf{z}_{i}\left(\mathbf{s}_{i}-\overline{\mathbf{s}}\right)^{T}=\mathbf{0}
+
+
+     The solution to this problem is found by centering sensitive features, fitting a
+     linear regression model to the non-sensitive features and reporting the residual.
+
+     The columns in :math:`S` will be dropped but the hyper parameter :math:`\alpha` does allow you to tweak
+     the amount of filtering that gets applied.
+
+     .. math::
+
+       X_{\text{tfm}} = \alpha X_{\text{filtered}} + (1-\alpha) X_{\text{orig}}
+    """
+
+    def __init__(self, sensitive_feature_ids, alpha=1, center=True):
```

Comment:
I don't think that we need or use `center` anywhere any more, let's drop.

Reply:
Actually, on the second thought I'm in favor of the second option:
```python
def __init__(self, *, sensitive_feature_ids, alpha=1):
```
because it makes our API more future-proof. For example, if we want to add `control_feature_ids` as we did in some of our other metrics.